src/test/java/org/example/springbootintro/book/BookRepositoryTest.java

```diff
@@ -0,0 +1,99 @@
+package org.example.springbootintro.book;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+import org.example.springbootintro.model.Book;
+import org.example.springbootintro.repository.book.BookRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+class BookRepositoryTest {
+    @Mock
+    private BookRepository bookRepository;
```

Comment:
So you are testing book repository by mocking book repository? Looks like it does not have any sense, you should make a real requests pre-setup database to test repo